packages/test/stochastic-test-utils/src/test/utils.ts

```diff
@@ -0,0 +1,116 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable no-bitwise */
+
+import { strict as assert } from "assert";
+
+export class Counter<T> {
+    private readonly choiceToCount = new Map<T, number>();
+
+    public increment(value: T): void {
+        this.choiceToCount.set(value, this.get(value) + 1);
+    }
+
+    public get(value: T): number {
+        return this.choiceToCount.get(value) ?? 0;
+    }
+
+    public entries(): Iterable<[T, number]> {
+        return this.choiceToCount.entries();
+    }
+
+    public values(): Iterable<T> {
+        return this.choiceToCount.keys();
+    }
+
+    public counts(): Iterable<number> {
+        return this.choiceToCount.values();
+    }
+}
+
+export function computeChiSquared<T>(weights: [T, number][], sampleCounts: Counter<T>): number {
+    const values = Array.from(sampleCounts.values());
+
+    assert.deepEqual(new Set(weights.map(([value]) => value)), new Set(values),
+        "'weights' must include all choices and all choices must have at least occurrence in 'sampleCounts'.");
+
+    if (weights.length === 1) {
+        const [value, weight] = weights[0];
+
+        assert.deepEqual(weights, [[value, 1.0]],
+            `With a single choice the associated weight must be 1.0, but got ${weight}.`);
+
+        return 0;
+    }
```

Comment:
Note: I added lines 35..47, which assert that we run enough iterations that we receive at least 1 sample of all possible choices, and handle the case where only 1 choice is expected.